conversion/conversion.go

```diff
@@ -0,0 +1,789 @@
+// Copyright 2020 Google LLC
```

Comment:
In general, files in go should be organized with the important stuff at the top. The goal is to make the file readable and easy to browse. That usually means:
- constants first
- key type definitions next (although often it makes sense to put them next to public functions that use them)
- then public functions (and relevant type definitions)
- and helper functions and other non-public definitions last (generally in order of importance)

What do you think Asheesh? At the very least, we should put public functions before helper/internal functions?

Reply:
marked with TODO